app/src/main/java/de/westnordost/streetcomplete/map/PinIcons.kt

```diff
@@ -59,5 +60,9 @@ import de.westnordost.streetcomplete.R
     when (map["tourism"]) {
         "information" -> return R.drawable.ic_pin_information
     }
+    if ((map.containsKey("addr:housenumber") or map.containsKey("addr:housename"))
+        and !map.containsAnyKey("amenity", "craft", "leisure", "office", "shop", "tourism")){
```

Comment:
This should probably not use [bitwise logic](https://kotlinlang.org/docs/basic-types.html#operations) `and`/`or` infix functions.


```suggestion
    if ((map.containsKey("addr:housenumber") || map.containsKey("addr:housename"))
        && !map.containsAnyKey("amenity", "craft", "leisure", "office", "shop", "tourism")){
```